src/Symfony/Component/Translation/MessageCatalogue.php

```diff
@@ -55,26 +55,44 @@ public function getDomains(): array
         return array_values($domains);
     }
 
-    public function all(string $domain = null): array
+    public function all(string $domain = null, ?string $sort = null): array
     {
+        $sort = $sort ? strtolower($sort) : null;
+
         if (null !== $domain) {
             // skip messages merge if intl-icu requested explicitly
             if (str_ends_with($domain, self::INTL_DOMAIN_SUFFIX)) {
-                return $this->messages[$domain] ?? [];
+                $messages = $this->messages[$domain] ?? [];
+            } else {
+                $messages = ($this->messages[$domain.self::INTL_DOMAIN_SUFFIX] ?? []) + ($this->messages[$domain] ?? []);
+            }
+
+            if ('desc' === $sort) {
+                krsort($messages);
+            } elseif ('asc' === $sort) {
+                ksort($messages);
             }
 
-            return ($this->messages[$domain.self::INTL_DOMAIN_SUFFIX] ?? []) + ($this->messages[$domain] ?? []);
+            return $messages;
         }
 
         $allMessages = [];
 
         foreach ($this->messages as $domain => $messages) {
             if (str_ends_with($domain, self::INTL_DOMAIN_SUFFIX)) {
                 $domain = substr($domain, 0, -\strlen(self::INTL_DOMAIN_SUFFIX));
-                $allMessages[$domain] = $messages + ($allMessages[$domain] ?? []);
+                $messages = $messages + ($allMessages[$domain] ?? []);
             } else {
-                $allMessages[$domain] = ($allMessages[$domain] ?? []) + $messages;
+                $messages = ($allMessages[$domain] ?? []) + $messages;
             }
+
+            if ('desc' === $sort) {
+                krsort($messages);
+            } elseif ('asc' === $sort) {
+                ksort($messages);
+            }
```

Comment:
Should the method throw an error if the value of sort is invalid? I like not being too strict and doing nothing when the value is different, but it's less clear if a typo is made (e.g. passing the value `acs` instead of `asc`)